shared/util/codeql/suppression/AlertSuppression.qll

```diff
@@ -72,6 +78,47 @@ module Make<SingleLineComment Comment> {
     ) {
       this.hasLocationInfo(filepath, startline, _, endline, endcolumn) and
       startcolumn = 1
+      or
+      exists(int cStartLine, int cStartColumn, int cEndLine, int cEndColumn |
+        this.hasLocationInfo(filepath, cStartLine, cStartColumn, cEndLine, cEndColumn) and
+        not exists(int c, Node n | c < cStartColumn |
+          n.hasLocationInfo(filepath, _, _, cStartLine, c) or
+          n.hasLocationInfo(filepath, cStartLine, c, _, _)
+        ) and
+        // when there is no column information, a location spans the whole line
+        startcolumn = 0 and
+        endcolumn = 0 and
+        startline = cEndLine + 1 and
+        endline = startline
+      )
+    }
+  }
+
+  private class CodeQlSuppressionComment extends SuppressionComment {
+    private string annotation;
+
+    CodeQlSuppressionComment() {
+      // match `codeql[...]` anywhere in the comment
+      annotation = this.(Comment).getText().regexpFind("(?i)\\bcodeql\\s*\\[[^\\]]*\\]", _, _) and
+      exists(string filepath, int cStartLine, int cStartColumn |
+        this.(Comment).hasLocationInfo(filepath, cStartLine, cStartColumn, _, _) and
+        not exists(int c, Node n | c < cStartColumn |
+          n.hasLocationInfo(filepath, _, _, cStartLine, c) or
+          n.hasLocationInfo(filepath, cStartLine, c, _, _)
+        )
+      )
+    }
+
+    override string getAnnotation() { result = "lgtm" + annotation.suffix(6) }
+
+    override predicate covers(
+      string filepath, int startline, int startcolumn, int endline, int endcolumn
+    ) {
+      this.hasLocationInfo(filepath, _, _, startline - 1, _) and
```

Comment:
Is it on purpose that you are using the endline of the comment to calculate which line is covered? 
It seems more natural to me to base the calculation off the startline of the comment.

Reply:
It's on purpose. At the moment it shouldn't matter too much because the comments are single line anyway. However, if we'd allow things like the following then the "next" line should be the line after the end of the comment.
```java
/*
codeql[java/xss]
*/
response.getOutputStream().write(request.getParam("name"))
```